engine-workspace/src/contract.rs

```diff
@@ -0,0 +1,393 @@
+    }
+
+    pub fn id(&self) -> AccountId {
+        self.contract.id()
+    }
+
+    pub fn root(&self) -> Account {
+        self.node.root()
+    }
+}
+
+impl From<(RawContract, Node)> for EngineContract {
+    fn from((contract, node): (RawContract, Node)) -> Self {
+        Self { contract, node }
+    }
+}
+
+/// Callable functions implementation.
+impl EngineContract {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(
+        &self,
+        chain_id: RawU256,
+        owner_id: AccountId,
+        upgrade_delay_blocks: u64,
+    ) -> CallNew {
+        let args = NewCallArgs::V2(NewCallArgsV2 {
+            chain_id,
+            owner_id,
+            upgrade_delay_blocks,
+        });
+
+        CallNew::call(&self.contract).args_borsh(args)
+    }
+
+    pub fn new_eth_connector(
+        &self,
+        prover_account: AccountId,
+        custodian_address: String,
+        metadata: FungibleTokenMetadata,
+    ) -> CallNewEthConnector {
+        CallNewEthConnector::call(&self.contract).args_borsh((
+            prover_account,
+            custodian_address,
+            metadata,
+        ))
+    }
+
+    pub fn ft_transfer(
+        &self,
+        receiver_id: &AccountId,
+        amount: U128,
+        memo: Option<String>,
+    ) -> CallFtTransfer {
+        CallFtTransfer::call(&self.contract)
+            .args_json(json!({ "receiver_id": receiver_id, "amount": amount, "memo": memo }))
+    }
+
+    pub fn ft_transfer_call(
+        &self,
+        receiver_id: &AccountId,
+        amount: U128,
+        memo: Option<String>,
+        msg: String,
+    ) -> CallFtTransferCall {
+        CallFtTransferCall::call(&self.contract).args_json(json!({
+           "receiver_id": receiver_id,
+           "amount": amount.0.to_string(),
+           "memo": memo,
+           "msg": msg,
+        }))
+    }
+
+    pub fn storage_deposit(
+        &self,
+        account_id: Option<AccountId>,
+        registration_only: Option<bool>,
+    ) -> CallStorageDeposit {
+        CallStorageDeposit::call(&self.contract)
+            .args_json(json!({ "account_id": account_id, "registration_only": registration_only}))
+    }
+
+    pub fn storage_withdraw(&self, amount: Option<U128>) -> CallStorageWithdraw {
+        CallStorageWithdraw::call(&self.contract).args_json(json!({ "amount": amount }))
+    }
+
+    pub fn storage_unregister(&self, force: Option<bool>) -> CallStorageUnregister {
+        CallStorageUnregister::call(&self.contract).args_json(json!({ "force": force }))
+    }
+
+    pub fn withdraw(&self, recipient_address: Address, amount: u128) -> CallWithdraw {
+        CallWithdraw::call(&self.contract).args_borsh((recipient_address, amount))
+    }
+
+    pub fn deposit(&self, raw_proof: Proof) -> CallDeposit {
+        CallDeposit::call(&self.contract).args_borsh(raw_proof)
+    }
+
+    pub fn set_eth_connector_contract_data(
+        &self,
+        prover_account: AccountId,
+        eth_custodian_address: String,
```

Comment:
Why it's `String`?

Reply:
Because of [this](https://github.com/aurora-is-near/aurora-engine/blob/develop/engine-types/src/parameters/connector.rs#L16).